### PR TITLE
View Participant Records

### PR DESCRIPTION
**Pull request by francisbouchard**

### CODE REVIEW

File: express-server/routes/participant.route.js [hunk 1]
```diff
@@ -156,6 +162,48 @@ router.post('/:pid/doc', (req, res) => {
                         if(err){
                             res.status(500).send(err);
                         } else {
+                            if(req.files.attachment.mimetype == "application/pdf"){
```
Comment (annarog) [inline express-server/routes/participant.route.js]: what if `req.files` is undefined? add some validation here
(this throws an error when testing)

Comment (francisbouchard) [reply express-server/routes/participant.route.js]: Good catch

**Comment by adriannadiaz** [comment]
Also I think the dependencies might need a little work, there are some conflicts and some that need manual install, but I would wait until you update your branch since I fixed some conflicts already in my previous PR

File: express-server/package.json [hunk 2]
```diff
@@ -20,10 +20,15 @@
...
 				"mongo": "^0.1.0",
 				"mongoose": "^5.0.0-rc0",
 				"passport": "^0.4.0",
-				"passport-local": "^1.0.0"
+				"passport-local": "^1.0.0",
+				"pdf-image": "^1.0.5",
+				"pdf2json": "^1.1.7",
+				"pdfjs-dist": "^2.0.385",
+				"read-chunk": "^2.1.0"
 		},
```
Comment (Ldhatter) [inline express-server/package.json]: What are the differences between / reasons for all these PDF dependencies?

**Comment by Ldhatter** [comment]
Bug: Integration with Uploading / Missing Functionality
----
When a user uploads a new note or a document, there is no alert saying the document was saved, nor does the modal close and refresh the participants profile page (ie: new notes and documents don't immediately display on the profile page). Probably best to pull from master first, before attempting to fix this.

**Comment by Ldhatter** [comment]
Bug : Integration with Uploading / Overwriting files
-----

When a user uploads a file, the system will overwrite files with the same file name.
Eg: If I store a file named health.pdf (some form) then I want to upload another file that just so happened to be also named health.pdf (but now this is a receipt) then the 2nd pdf would have overwritten the 1st pdf. 

Possible solution? Create subfolders in the participants folder and name this subfolders with a unique identifier that matches up with that Document and Note mongo doc entry?

File: express-server/package.json (hunk 2 above)
Comment (Ldhatter) [reply express-server/package.json]: Oh i didnt see adrianna's comment. sorry for being redundant

**Comment by Ldhatter** [comment]
Bug [Critical] : Saving and Viewing Records!
-----

Saving: i see the file is saved on the server side, but then in mongo, there's nothing saved!
Viewing: Since it doesn't save, I can't view any records (documents or notes).

File: express-server/routes/participant.route.js [hunk 3]
```diff
@@ -200,6 +209,48 @@ router.post('/:pid/doc', (req, res) => {
                         if(err){
                             res.status(500).send(err);
                         } else {
+                            if(req.files.attachment.mimetype && req.files.attachment.mimetype == "application/pdf"){
```
Comment (adriannadiaz) [inline express-server/routes/participant.route.js]: This is copy-paste programming. Find a way to make it modular and do not use the copy-paste technique.

**Comment by adriannadiaz** [comment]
Also provide high-resolution thumbnails for people with higher-resolution screens so output at 1200*1600 and display at 600 x 800 ? Or maybe they should be thumbnails and displayed at half size ?   I do find it useful to have the letters readable...

**Comment by adriannadiaz** [comment]
Also I am not confortable with the thumbnails being dumped on Mongo, is it that expensive to render them ? If so maybe we should limit to 1-4 per document as I wrote before.

**Comment by annarog** [comment]
@francisbouchard I see Notes and Documents duplicated inside the participant profile. I think something went wrong during the merge from master?
(@Ldhatter I guess that's why you can't view records? I'm able to save/view them but it only displays in the second occurrence of "Notes")
Also, Delete Document/Note works in the first set of Documents/Notes, but it doesn't work in the second set.

screen shot 2018-03-07 at 3 06 55 pm

**Comment by Ldhatter** [comment]
@annarog  i made that review _before_ the merge from master, so i dont know what he was missing beforehand but it wasn't even displaying anything when i checked out the branch. but since the merge, the bug you found should definitely be dealt with.

**Comment by francisbouchard** [comment]
@annarog This was fixed in the last commit.

**Comment by francisbouchard** [comment]
@adriannadiaz if you don't want the thumbnails to be stored in mongo I am not sure where else to put them... What makes you uncomfortable exactly? They are small images (~30kb)

**Comment by francisbouchard** [comment]
@adriannadiaz  the issue specifically asked for 600X800 images...

**Comment by adriannadiaz** [comment]
There is still no frontend testing. @francisbouchard @Ldhatter

**Comment by francisbouchard** [comment]
@adriannadiaz Not really sure what I should be testing. Most of my work was done in the backend. The only major addition to the frontend was the image tag.

**Pull request closed without merging**